subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/TaskNodeCodec.kt

```diff
@@ -252,7 +253,7 @@ sealed class RegisteredProperty {
         val optional: Boolean,
         val filePropertyType: InputFilePropertyType,
         val behavior: InputBehavior,
-        val fileNormalizer: Class<out FileNormalizer>?,
+        val normalizer: Normalizer?,
```

Comment:
Why we need `?` here?

Reply:
We can remove this later probably.

Reply:
Actually, we cannot, because we need to distinguish between "no normalization" and "explicitly absolute normalization" cases for cacheable tasks (we penalize the first, accept the second).